jwst/transforms/jwst_types.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+
+"""
+Defines a ``JWSTTransformType`` used by jwst pipeine transform types.
+All types are added automatically to ``_jwst_types`` and the JWSTExtension.
+
+"""
+import six
+
+from asdf.asdftypes import ExtensionTypeMeta
+from astropy.io.misc.asdf.tags.transform.basic import TransformType
+from astropy.io.misc.asdf.types import AstropyTypeMeta
+
+__all__ = ['JWSTTransformType']
+
+
+_jwst_types = set()
+
+
+class JWSTTypeMeta(AstropyTypeMeta):
+    """
+    Keeps track of `JWSTType` subclasses that are created so that they can
+    be stored automatically by astropy extensions for ASDF.
+    """
+    def __new__(mcls, name, bases, attrs):
+        cls = super(JWSTTypeMeta, mcls).__new__(mcls, name, bases, attrs)
+        # Classes using this metaclass are automatically added to the list of
+        # astropy extensions
+        if cls.organization == 'stsci.edu' and cls.standard == 'jwst_pipeline':
```

Comment:
This `if` statement is probably not necessary in this case. It is useful in the astropy extension because Astropy may theoretically implement tags from multiple standards (e.g. from the ASDF standard and from Astropy itself). There's no harm in leaving it, though.